.ci/test.sh

```diff
@@ -64,7 +64,7 @@ if [[ $TASK == "lint" ]]; then
     echo "Linting R code"
     Rscript ${BUILD_DIRECTORY}/.ci/lint_r_code.R ${BUILD_DIRECTORY} || exit -1
     echo "Linting C++ code"
-    cpplint --filter=-build/c++11,-build/include_subdir,-build/header_guard,-whitespace/line_length --recursive ./src ./include ./R-package || exit -1
+    cpplint --filter=-build/c++11,-build/include_subdir,-build/header_guard,-whitespace/line_length --recursive ./src ./include ./R-package ./swig || exit -1
```

Comment:
what is the reason for this change? We use linters like this to keep the style of the project's code similar across all parts of the repo, so that developers working on LightGBM don't need to change styles from file to file. I do actually prefer the style you've changed to in `StringArray.hpp` but that is irrelevant...we use linters because we believe that the value in *consistency* of the code style is more important than choosing any particular style that certain people have a personal preference for.

Can you please revert this unless it is somehow related to the *correctness* of the code?

Similarly, it seems like you've made some style-only changes in this PR, for example:

![image](https://user-images.githubusercontent.com/7608904/111535462-2219cf80-8737-11eb-85cb-882d001c862f.png)

Those increase the size of the diff that reviewers have to review without actually changing the way the code functions, and we'd appreciate if you avoided such changes. Can you please revert style-only changes?

If you've already talked about this with @StrikerRUS and @StrikerRUS was fine with this modification to which code is checked by `cpplint`, feel free to ignore this comment. I didn't click through and re-open all of the previous conversations that were marked "resolved".

Reply:
The C++ code in the swig folder was far from following our standards so StrikerRUS asked me to enable linting there and address any warnings.

Reply:
OHHHH I read too fast and thought you were adding an **exclusion**, not **inclusion** for `swig/`. Ok, carry on :joy: